app/script/entity/message/Asset.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Wire
+ * Copyright (C) 2017 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.entity = z.entity || {};
+
+z.entity.Asset = class Asset {
+  constructor(id) {
+    this.id = id;
+    this.key = '';
+    this.type = '';
+  }
+
+  /**
+   * Check if asset is a medium image.
+   *
+   * @returns {Boolean} Is asset of type medium image
+   */
+  is_image() {
+    return this.type === z.assets.AssetType.IMAGE;
+  }
+
+  /**
+   * Check if asset is a text.
+   *
+   * @returns {Boolean} Is asset of type text
+   */
+  is_text() {
+    return this.type === z.assets.AssetType.TEXT;
+  }
+
+  /**
+   * Check if asset is a file.
+   *
+   * @returns {Boolean} Is asset of type file
+   */
+  is_file() {
+    return this.type === z.assets.AssetType.FILE && !this.is_video() && !this.is_audio();
+  }
+
+  /**
+   * Check if asset is a location.
+   *
+   * @returns {Boolean} Is asset of type location
+   */
+  is_location() {
+    return this.type === z.assets.AssetType.LOCATION;
+  }
+
+  /**
+   * Check if asset is a video.
+   *
+   * @returns {Boolean} Is asset of type video
+   */
+  is_video() {
+    const is_video_asset = (this.type === z.assets.AssetType.FILE) && (this.file_type != null ? this.file_type.startsWith('video') : undefined);
```

Comment:
`const is_video_asset = this.type === z.assets.AssetType.FILE && this.file_type && this.file_type.startsWith('video');`  seems easier to read to me